Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -44,6 +44,115 @@ static wstring __GetUserDefaultLocaleName() {
 }
 
 /**
+ * Private helper that creates an uppercase copy of a CFString
+ */
+CFStringRef __CFStringCreateUppercaseCopy(CFStringRef string, CFLocaleRef locale) {
+    CFMutableStringRef ret = CFStringCreateMutableCopy(nullptr, CFStringGetLength(string), string);
+    CFStringUppercase(ret, locale);
+    return ret;
+}
+
+/**
+ * Private helper that creates the IDWriteFontList for a given font family name
+ */
+HRESULT __DWriteGetFontListForFamilyName(CFStringRef familyName, IDWriteFontList** outFontList) {
+    ComPtr<IDWriteFactory> dwriteFactory;
+    RETURN_IF_FAILED(DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(IDWriteFactory), &dwriteFactory));
```

Comment:
Not DWrite factories.  These aren't same as winrt activation objects. [https://msdn.microsoft.com/en-us/library/windows/desktop/dd368040(v=vs.85).aspx](https://msdn.microsoft.com/en-us/library/windows/desktop/dd368040(v=vs.85).aspx). This function creates a DirectWrite factory object that is used for subsequent creation of individual DirectWrite objects. DirectWrite factory contains internal state data such as font loader registration and cached font data. In most cases it is recommended you use the shared factory object, because it allows multiple components that use DirectWrite to share internal DirectWrite state data, and thereby reduce memory usage. However, there are cases when it is desirable to reduce the impact of a component, such as a plug-in from an untrusted source, on the rest of the process, by sandboxing and isolating it from the rest of the process components. In such cases, it is recommended you use an isolated factory for the sandboxed component. 

---
In reply to: [87901335](https://github.com/Microsoft/WinObjC/pull/1366#discussion_r87901335) [](ancestors = 87901335,87900429)